src/createUseConfig.ts

```diff
@@ -0,0 +1,20 @@
+import { Config, InjectedProps, NamespacedUseConfigReturnType } from "./types";
+import { useCallback } from "react";
+import { useWatchLocalstorageEvents, capitalize } from "./utils";
+
+export function createUseConfig<TSchema extends Record<string, Config>, Namespace extends string>(
+  props: InjectedProps<TSchema, Namespace>,
+) {
+  return () => {
+    const localstorageDep = useWatchLocalstorageEvents(props.storage, props.localOverride);
```

Comment:
For the sake of consistency with the `window` propery name, let's name it camelcase here as well 
```suggestion
    const localStorageDependency = useWatchLocalstorageEvents(props.storage, props.localOverride);
```